crypto/crypto_aesctr.c

```diff
@@ -42,6 +42,20 @@ crypto_aesctr_init(const struct crypto_aes_key * key, uint64_t nonce)
 	return (NULL);
 }
 
+/* Generate a block of cipherstream. */
+static inline void
```

Comment:
The `inline` keyword shouldn't do anything here on modern compilers.  Do you see any performance difference by removing it?

Reply:
Very slight difference in gcc 9.3.0 with `-O2` and buffer sizes of 16 and higher.  No difference in clang or gcc with `-O3 -march=native`.

- blocksize 16: slightly faster without `inline`
- blocksize 32: equal
- blocksize 64 to 1024: slower without `inline` (but still within 1%).

![noinline](https://user-images.githubusercontent.com/222157/100272635-59f2d200-2f10-11eb-93b8-bfa92b36e350.png)

I thought this was weird, so I re-ran the gcc tests in the opposite order, with 7 repetitions instead of 3, and with gcc 10.2.1 instead of gcc 9.3.0.  Same results (although this time there's enough data to label a few points as outliers; those are circles).

![again](https://user-images.githubusercontent.com/222157/100275255-7ee94400-2f14-11eb-8098-e67d1fddcc8d.png)


In case you're wondering "why is there an 'outlier' in 1024 inline that's drawn right on top of the other data", here's the raw figures:
```
$ grep 32768 perftest-crypto_aesctr-gcc10-inline.txt
32768	1024	0.096586	347.405
32768	1024	0.096625	347.264
32768	1024	0.096625	347.264
32768	1024	0.096613	347.308
32768	1024	0.096625	347.264
32768	1024	0.096627	347.257
32768	1024	0.096624	347.268
```

(this is after tweaking the perftest system to show 3 decimal places for speed, instead of rounding to 1)

Looks very stable, other than perhaps a slight "warm up" penalty for the first one.

Anyway, it appears that `inline` does have a reproducible effect with `gcc`, but IMO it's not big enough to worry about.  I'm happy to remove it if you prefer that.

(FWIW, we also have `static inline int doevent(struct eventrec * r`, which is why I used `inline` here.)

Reply:
Ok, if it helps, keep it.  I was just surprised that it did anything at all -- I thought compilers generally made their own decisions about what should be inlined, but maybe gcc is taking something from the hint.

Reply:
I agree that it's surprising that `inline` helps, and it's not something that I tested earlier.